### ISSUE TITLE
Training a [yolov5su.pt] model seems to download the [yolov8n.pt] model? (8.0.55)

### ISSUE THREAD

**Issue by JustasBart** [ultralytics/ultralytics#1570]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi guys,

Am I missing something here or not doing something right but why does executing the following:

```
yolo detect train \
data=*DATASET*.yaml \
model=yolov5su.pt \
epochs=500 \
patience=250 \
imgsz=768 \
rect=True \
device=0 \
batch=-1 \
plots=True \
scale=0.7 \
flipud=0.5 \
fliplr=0.5 
```

Result in:
![image](https://user-images.githubusercontent.com/40023722/226957779-16774da9-7337-475e-9086-d2a2a1f1ff9e.png)

But then also a little bit more in:
![image](https://user-images.githubusercontent.com/40023722/226958210-e570cafd-c532-4510-8ba7-1d76bcd50176.png)

What is this training on now?
Is it the **yolov5su.pt** like I would expect?
Or is is the **yolov8n.pt** because something didn't quite go right here?

Was it meant to be **model='yolov5su.pt'** or something?

Thanks :rocket: 

### Additional

The file size sort of looks like that of a **Small** size network:

![image](https://user-images.githubusercontent.com/40023722/226959601-bfa878bd-48d1-4dfe-ad8d-8fd9d701241f.png)

That opposed to a **Nano** size network, but that could be misleading also due to the optimiser...

**Comment by shayandaneshvar** [comment]
I have read a bit of the code. It seems that yolov8n.pt is downloaded every time as a check.
But I have the same question too. if we choose yolov8s.yaml as the model and set the pretrained switch to True, will it actually use the pretrained yolov8s or not?
I am saying this because I am getting slightly worse results when comparing yolov8s with yolov5s!!!
In fact no improvement in terms of accuracy between the small models.

**Comment by glenn-jocher** [comment]
@JustasBart 👋 Hello! Thanks for asking about **training checkpoints**.  I have YOLOv5 info, below but the same applies to YOLOv8.

YOLOv8n is used for AMP checks only.

YOLOv5 🚀 checkpoints should be about 4X the size of final trained checkpoints, as they carry not just a FP16 model, but a FP16 [EMA](https://www.tensorflow.org/api_docs/python/tf/train/ExponentialMovingAverage) and an FP32 optimizer of the same size as the model (each model parameter has its own FP32 gradient saved within the optimizer). Final checkpoints contain only an FP16 model, with the EMA and optimizer both stripped after the final epoch of training:
https://github.com/ultralytics/yolov5/blob/b4a29b5a8d63a8c2d4a8929942b44e8969c5dddd/train.py#L423-L425

The `strip_optimizer()` function updates the checkpoint dictionary value for `model`, replacing it with `ema`, and sets `ema` and `optimizer` keys to `None`, which will reduce the checkpoint size by 3/4.
https://github.com/ultralytics/yolov5/blob/b4a29b5a8d63a8c2d4a8929942b44e8969c5dddd/utils/general.py#L741-L755

Final trained checkpoints after `strip_optimizer()` should match the README table sizes below.
<p align="center"><img width="700" alt="YOLOv5 Models" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/model_comparison.png"></p>

You can also run `strip_optimizer()` manually on any checkpoint to convert it into finalized weights ready to train a new model:
```python
from utils.general import strip_optimizer

strip_optimizer('path/to/best.pt')
!python train.py --weights path/to/best.pt  # use best.pt to train a new model
```

### Model Exports

Exported models may be larger or smaller than their PyTorch starting points depending on their quantization. i.e. FP16 models will be the same size, FP32 models will be 2X the size, and INT8 models will be 0.5X the size. See YOLOv5 [Export tutorial](https://github.com/ultralytics/yolov5/issues/251) for quantization options on model export.

Good luck 🍀 and let us know if you have any other questions!

**Comment by JustasBart** [comment]
Hi @glenn-jocher the issue here wasn't on stripping the optimiser, it was just confusing that by training a **yolov5su.pt** model it always downloads the **yolov8n.pt** model for no apparent reason.

Afte stripping the optimiser it became clear that all was well in the end:

![image](https://user-images.githubusercontent.com/40023722/227195962-7636b2b2-b5b8-41fa-ae8f-b4660152ff57.png)

![image](https://user-images.githubusercontent.com/40023722/227196010-3234bd2f-bf4d-482a-8416-fb33051775d1.png)

But the original thread was just based on why does this happen:

![image](https://user-images.githubusercontent.com/40023722/227196211-f43c96ef-aedb-4ba3-9c0b-2407b6192984.png)

Even though **yolov8n.pt** is not actually used in any way.

**Comment by glenn-jocher** [comment]
Also, it is not always downloaded, it is only downloaded if it is not found locally.

**Comment by JustasBart** [comment]
@glenn-jocher Oh, I see! That makes sense then, was just confused as per why that might be happening...

**Comment by LouisJULIEN** [comment]
I found that very confusing as well, I was wondering if I hadn't misconfigured something. An additional log as explanation could be great.
Thanks for your answer Glenn

**Comment by glenn-jocher** [comment]
@LouisJULIEN i understand the confusion regarding the download of `yolov8n.pt` during training, and appreciate your suggestion for additional logging to clarify this step! 😊 It's always our aim to make the training process as transparent and user-friendly as possible. 

The download is part of ensuring the environment supports AMP (Automatic Mixed Precision) training — a quick check using a known model. We'll look into making this clearer in future versions with an additional log message. 

Thanks for your feedback! If you have any more questions or suggestions, feel free to share.
